### PR TITLE
chrome: don't nest pw:api steps

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

**Review by dgozman — APPROVED** [review]

File: src/client/types.ts [hunk 1]
```diff
@@ -27,7 +26,8 @@ export interface Logger {
 }
 
 export interface ClientSideInstrumentation {
-  onApiCall(stackTrace: ParsedStackTrace): ((error?: Error) => void) | undefined;
+  onApiCallBegin(apiCall: string): { userObject: any };
```
Comment (dgozman) [inline src/client/types.ts:29]: I think we can replace `{ userObject: any }` in both methods with `any` and it should be equivalent?

File: tests/playwright-test/reporter.spec.ts [hunk 2]
```diff
@@ -328,8 +330,8 @@ test('should report api step failure', async ({ runInlineTest }) => {
     `%% end {\"title\":\"Before Hooks\",\"category\":\"hook\",\"steps\":[{\"title\":\"browserContext.newPage\",\"category\":\"pw:api\"}]}`,
     `%% begin {\"title\":\"page.setContent\",\"category\":\"pw:api\"}`,
     `%% end {\"title\":\"page.setContent\",\"category\":\"pw:api\"}`,
-    `%% begin {\"title\":\"page.click\",\"category\":\"pw:api\"}`,
-    `%% end {\"title\":\"page.click\",\"category\":\"pw:api\",\"error\":{\"message\":\"page.click: Timeout 1ms exceeded.\\n=========================== logs ===========================\\nwaiting for selector \\\"input\\\"\\n============================================================\",\"stack\":\"<stack>\"}}`,
+    `%% begin {\"title\":\"page.click(input)\",\"category\":\"pw:api\"}`,
+    `%% end {\"title\":\"page.click(input)\",\"category\":\"pw:api\",\"error\":{\"message\":\"page.click(input): Timeout 1ms exceeded.\\n=========================== logs ===========================\\nwaiting for selector \\\"input\\\"\\n============================================================\",\"stack\":\"<stack>\"}}`,
```
Comment (dgozman) [inline tests/playwright-test/reporter.spec.ts:334]: Where does the `input` comes from to the error message? I guess I missed that in the code, but it seems like a lot more of our tests should be rebased then.

**Review by dgozman — APPROVED** [review]

File: src/client/types.ts (hunk 1 above)
Comment (pavelfeldman) [reply src/client/types.ts:29]: the instrumentation end uses the fact that it is an envelope...

Comment (pavelfeldman) [reply src/client/types.ts:29]: And more importantly, this documents it, right?

File: tests/playwright-test/reporter.spec.ts (hunk 2 above)
Comment (pavelfeldman) [reply tests/playwright-test/reporter.spec.ts]: It does not, fixed.

**Pull request merged**